test/tstCompileOnlyAccessTraits.cpp

```diff
@@ -58,17 +59,41 @@ struct ArborX::Traits::Access<LegacyAccessTraits, Tag>
   static Point get(LegacyAccessTraits, int) { return {}; }
 };
 
+template <class V, class Tag>
+using deduce_type_t =
+    decltype(ArborX::AccessTraits<V, Tag>::get(std::declval<V>(), 0));
+
 void test_access_traits_compile_only()
 {
   Kokkos::View<ArborX::Point *> p;
   Kokkos::View<float **> v;
   check_valid_access_traits(PrimitivesTag{}, p);
   check_valid_access_traits(PrimitivesTag{}, v);
 
+  auto p_with_indices = ArborX::Experimental::attach_indices(p);
+  check_valid_access_traits(PrimitivesTag{}, p_with_indices,
+                            ArborX::Details::DoNotCheckGetReturnType());
+  static_assert(
+      std::is_same_v<deduce_type_t<decltype(p_with_indices), PrimitivesTag>,
+                     ArborX::PairValueIndex<ArborX::Point, unsigned>>);
+
+  auto p_with_indices_long = ArborX::Experimental::attach_indices<long>(p);
+  static_assert(std::is_same_v<
+                deduce_type_t<decltype(p_with_indices_long), PrimitivesTag>,
+                ArborX::PairValueIndex<ArborX::Point, long>>);
+
   using NearestPredicate = decltype(ArborX::nearest(ArborX::Point{}));
   Kokkos::View<NearestPredicate *> q;
   check_valid_access_traits(PredicatesTag{}, q);
 
+  auto q_with_indices = ArborX::Experimental::attach_indices<long>(q);
+  check_valid_access_traits(PredicatesTag{}, q_with_indices);
+  using predicate = deduce_type_t<decltype(q_with_indices), PredicatesTag>;
+  static_assert(
+      std::is_same_v<
+          std::decay_t<decltype(ArborX::getData(std::declval<predicate>()))>,
```

Comment:
Why decay?

Reply:
`getData` return const ref.

Reply:
And we don't define our own `remove_cvref` trait right?  That one is C++20.